rrdd/rrdd_server.ml

```diff
@@ -416,29 +417,51 @@ module Plugin = struct
 
 		(* Helper function to find plugin info from a uid. *)
 		let find ~(uid: P.uid) : plugin =
-			Mutex.execute registered_m
-				(fun () -> Hashtbl.find registered uid)
+			try
+				Mutex.execute registered_m (fun () -> Hashtbl.find registered uid)
+			with
+				Not_found ->
+					error "failed to find plugin %s" (P.string_of_uid uid);
+					raise Rrd_protocol.Read_error
```

Comment:
The problem is, `get_payload` must either return a value or an exception. A `Skipped` exception or similar might be better.